File: backend/app/schemas/enums.py
"""全局枚举值。

SSOT：`../../../开发文档/fastapi_prompt_schema.md` 格子 3「全局枚举值一览」+
      `../../../开发文档/SQL_prompt_schema.md` 表字段 CHECK 约束。
前后端两套 schema（Python StrEnum / TypeScript union）保持字面值严格一致。
"""
from __future__ import annotations

from enum import StrEnum


class AssetCategory(StrEnum):
    US_STOCK = "美股"
    HK_STOCK = "港股"
    CN_STOCK = "中概股"
    CRYPTO = "加密货币"
    GOLD = "黄金"
    US_TREASURY = "美国国债"
    CN_TREASURY = "中国国债"


class AssetCurrency(StrEnum):
    """资产币种（6 种）。用于 assets / transactions。"""

    CNY = "CNY"
    USD = "USD"
    HKD = "HKD"
    EUR = "EUR"
    GBP = "GBP"
    CHF = "CHF"


class CashOrIncomeCurrency(StrEnum):
    """现金账户/收入/用户目标币种（2 种）。"""

    CNY = "CNY"
    USD = "USD"


class TransactionType(StrEnum):
    BUY = "buy"
    SELL = "sell"
    CLOSE = "close"


class IncomeCategory(StrEnum):
    LABOR = "纯劳动收入"
    CODE_MEDIA = "代码&自媒体收入"
    CAPITAL = "资本收入"

File: frontend/src/types/enums.ts
export type TransactionType = 'buy' | 'sell' | 'close'

export type AssetCategory =
  | '美股'
  | '港股'
  | '中概股'
  | '加密货币'
  | '黄金'
  | '美国国债'
  | '中国国债'

export type AssetCurrency = 'CNY' | 'USD' | 'HKD' | 'EUR' | 'GBP' | 'CHF'

export type CashOrIncomeCurrency = 'CNY' | 'USD'

export type IncomeCategory = '纯劳动收入' | '代码&自媒体收入' | '资本收入'

export type ErrorCode =
  | 'INVALID_SYMBOL'
  | 'DUPLICATE_ASSET'
  | 'INSUFFICIENT_POSITION'
  | 'ASSET_NOT_FOUND'
  | 'ACCOUNT_NOT_FOUND'
  | 'EXCHANGE_RATE_MISSING'
  | 'PRICE_MISSING'
  | 'EXTERNAL_SOURCE_FAILED'
  | 'INVALID_EMAIL_FORMAT'
  | 'EMAIL_SEND_FAILED'
  | 'NO_ACTIVE_ASSETS'
  | 'VALIDATION_ERROR'
  | 'INTERNAL_ERROR'
